ios/Classifier/VCPTaxonomy.m

```diff
@@ -133,32 +136,33 @@ - (VCPPrediction *)inflateTopPredictionFromClassification:(MLMultiArray *)classi
 // following
 // https://github.com/inaturalist/inatVisionAPI/blob/multiclass/inferrers/multi_class_inferrer.py#L136
 - (NSDictionary *)aggregateScores:(MLMultiArray *)classification currentNode:(VCPNode *)node {
-    if (node.children.count > 0) {
-        // we'll populate this and return it
-        NSMutableDictionary *allScores = [NSMutableDictionary dictionary];
-
-        for (VCPNode *child in node.children) {
-            NSDictionary *childScores = [self aggregateScores:classification currentNode:child];
-            [allScores addEntriesFromDictionary:childScores];
-        }
+    NSMutableDictionary *allScores = [NSMutableDictionary dictionary];
 
+    if (node.children.count > 0) {
         float thisScore = 0.0f;
         for (VCPNode *child in node.children) {
-            thisScore += [allScores[child.taxonId] floatValue];
+            NSDictionary *childScores = [self aggregateScores:classification currentNode:child];
+            NSNumber *childScore = childScores[child.taxonId];
+            
+            if ([childScore floatValue] > self.taxonomyRollupCutoff) {
+                [allScores addEntriesFromDictionary:childScores];
+                thisScore += [childScore floatValue];
+            }
         }
-
         allScores[node.taxonId] = @(thisScore);
 
-        return [NSDictionary dictionaryWithDictionary:allScores];
     } else {
         // base case, no children
         NSAssert(node.leafId, @"node with taxonId %@ has no children but also has no leafId", node.taxonId);
         NSNumber *leafScore = [classification objectAtIndexedSubscript:node.leafId.integerValue];
         NSAssert(leafScore, @"node with leafId %@ has no score", node.leafId);
-        return @{
-            node.taxonId: leafScore
-        };
+        
+        if ([leafScore floatValue] > self.taxonomyRollupCutoff) {
+            allScores[node.taxonId] = leafScore;
+        }
     }
+    
+    return [allScores copy];
```

Comment:
If I heard correctly, the changes in this method led to an ~7x speedup. Since most of the apparent work is simple data structures rather than computation, it would be nice to benchmark the Objective-C against a Swift version.

Reply:
We have a [swift implementation on the todo list](https://github.com/inaturalist/vision-camera-plugin-inatvision/issues/4) but we don't have one finished to benchmark. Do you want this done before the PR is accepted or is it OK to add this benchmarking note to that issue?

Reply:
Definitely not a blocker for this PR, just me being unfamiliar with the todo list and speculating out loud.